### PR TITLE
When selecting font, attempt to use system-based font before internally supplied version

### PR DESCRIPTION
**Pull request by ddb4github**
* Delete buildin DejaVu font files
* Use OS provided DejaVu font files, append default for RHEL/CentOS/SLES/Ubuntu

### CODE REVIEW

**Comment by TheWitness** [comment]
@netniV, what do you think?  Does not consider Windows for sure.

File: lib/rrd.php [hunk 1]
```diff
@@ -3592,7 +3592,13 @@ function rrdtool_create_error_image($string, $width = '', $height = '') {
...
 
 	if ($config['cacti_server_os'] == 'unix') {
-		$font_file = '/usr/share/fonts/dejavu/DejaVuSans.ttf';
+		if (file_exists('/usr/share/fonts/dejavu/DejaVuSans.ttf')) {
+			$font_file = '/usr/share/fonts/dejavu/DejaVuSans.ttf'; //RHEL/CentOS
+		} else if (file_exists('/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf')) {
+			$font_file = '/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf'; //Ubuntu
+		} else {
+			$font_file =  '/usr/share/fonts/truetype/DejaVuSans.ttf'; //SLES
+		}
```
Comment (netniV) [inline lib/rrd.php:3601]: Again see above.

**Review by netniV — CHANGES_REQUESTED** [review]
I think using the system supplied font is useful, though if it is behind our version of the font, this may cause issues.  However, since the packaged versions of cacti will remove fonts in favour of the system supplied ones, I think people like @paulgevers would cheer this change on.  It doesn't apply to windows currently, as we seem to hard code to the Arial font on windows, though we could use a font_paths array and a font_names array to combine them into a search path.

Also, if this is related to an issue, I don't recall seeing one and there should be an issue tracker number and a CHANGELOG

File: lib/functions.php [hunk 2]
```diff
@@ -5464,8 +5464,15 @@ function get_classic_tabimage($text, $down = false) {
 	$wrapsize = 12;
 
 	if (file_exists($config['base_path'] . '/images/' . $images[$down])) {
+		if (file_exists('/usr/share/fonts/dejavu/')) {
+			$font_path = '/usr/share/fonts/dejavu/'; //RHEL/CentOS
+		} else if (file_exists('/usr/share/fonts/truetype/dejavu/')) {
+			$font_path = '/usr/share/fonts/truetype/dejavu/'; //Ubuntu
+		} else {
+			$font_path =  '/usr/share/fonts/truetype/'; //SLES
+		}
```
[the remark this answers is not in the document]
Comment (ddb4github) [reply lib/functions.php:5473]: Copilot auto output dozen of DejaVu paths, all of them do not match known RHEL/SLES/Ubuntu paths. I'm not sure which one for which OS. 
So I'll combine RHEL/SLES/Ubuntu path in array only.
```php
//Copilot completion
$dejavu_paths = array (
	'C:/Program Files/DejaVu/',
	'C:/Program Files (x86)/DejaVu/',
	'/usr/local/dejavu/',
	'/usr/local/openwin/local/share/dejavu/',
	'/usr/local/openwin/share/dejavu/',
	'/usr/local/share/dejavu/',
	'/usr/local/X11R6/local/share/dejavu/',
	'/usr/local/X11R6/share/dejavu/',
	'/usr/openwin/local/share/dejavu',
	'/usr/openwin/share/dejavu/',
	'/usr/share/dejavu/',
	'/usr/X11R6/local/share/dejavu/',
	'/usr/X11R6/share/dejavu/'
);
//RHEL/SLES/Ubuntu known path
$dejavu_paths = array (
	'/usr/share/fonts/dejavu/',
	'/usr/share/fonts/truetype/',
	'/usr/share/fonts/truetype/dejavu/'
);

**Comment by TheWitness** [comment]
Let's make sure Windoz is covered.

**Comment by netniV** [comment]
I would also like to keep the currently included font as backup.

**Comment by TheWitness** [comment]
Jing, make the Cacti supplied Dejavu don't the last choice in the array.  Did you check that windows actually includes this font.  Maybe not by default.  I'll look after I'm fully alert.

**Comment by TheWitness** [comment]
Okay, I must not be awake.  This is perfect now. 👍

**Comment by TheWitness** [comment]
Everything looks right.  Thanks for your contribution Jing!

**Pull request merged**